config/config.go

```diff
@@ -170,3 +173,192 @@ func defaultRunPath() string {
 	}
 	return filepath.Join(homeDir, ".earthly/run")
 }
+
+// UpsertConfig adds or modifies the key to be the specified value.
+// This is saved to disk in your earthly config file.
+func UpsertConfig(config []byte, path, value string) ([]byte, error) {
+	base := &yaml.Node{}
+	yaml.Unmarshal(config, base)
+
+	if base.IsZero() {
+		// Empty file, or a simple comment results in a null document.
+		// Not handled well, so manufacture somewhat acceptable document
+		fullDoc := string(config) + "\n---"
+		yaml.Unmarshal([]byte(fullDoc), base)
+		base.Content = []*yaml.Node{{Kind: yaml.MappingNode}}
+	}
+
+	pathParts := splitPath(path)
+
+	err := validatePath(reflect.TypeOf(Config{}), pathParts)
```

Comment:
Uses the `yaml` tags on our existing structs to validate. We can just add config to the structs like normal and it should magically also be writeable.